Source/WebCore/dom/Element.cpp

```diff
@@ -265,6 +265,8 @@ Element::~Element()
         if (auto* map = elementRareData()->attributeStyleMap())
             map->clearElement();
     }
+
+    document().removeElementWithExplicitLanguageMatchingDocumentElement(*this);
```

Comment:
This isn't quite right. An element can move from one document to another.
Please do this in removedFromAncestor instead.
Also, we need to insert the element into the map when an element with existing lang attribute gets inserted into document.

Reply:
I think it will be simpler not to use the optimization for elements outside the document. Otherwise we really need to deal with the adoption of the elements to get this right, rather than just when it's inserted and removed.